src/irmin-pack/inode.ml

```diff
@@ -335,8 +345,10 @@ struct
         | Truncated -> (
             function Broken h -> h | Intact ptr -> Lazy.force ptr.hash)
 
-      let target : type ptr. cache:bool -> ptr layout -> ptr -> ptr t =
-       fun ~cache layout ->
+      let target :
+          type ptr.
+          cache:bool -> force:bool -> string -> ptr layout -> ptr -> ptr t =
+       fun ~cache ~force context layout ->
```

Comment:
This `force` trick to compute the merkle proof also includes the inodes that were loaded but not modified, is this intended?

You should maybe raise `Dangling_hash` when `force = true` and `{ target = Lazy_loaded entry }`

Reply:
the goal is to never call `find` when building a proof as we want the minimal inode/tree which can be used to verify a sequence of operations later on.

I am not sure to understand your remark (as I am not sure to understand exactly what this inode code is doing 😓 ). I am not sure what's the difference? Normally, you can just "touch" all the keys that the sequence of operation does to get your proof. Are you suggesting we can do something better?

E.g. see https://github.com/samoht/irmin/blob/2.8-merke-proofs/src/irmin/tree.ml#L1880-L1883 on we do this currently. But I'm open to a better way :-)

Reply:
> the goal is to never call find when building a proof as we want the minimal inode/tree which can be used to verify a sequence of operations later on.

I see now! Thanks